src/librustc_mir/borrow_check/mod.rs

```diff
@@ -2208,3 +2277,15 @@ impl ContextKind {
         }
     }
 }
+
+#[derive(Debug)]
+crate struct Upvar {
```

Comment:
Yeah I think I would prefer to see `closure_upvars(def_id, n) -> Upvar` that returns this information "nicely collected"....  (or `closure_upvars(def_id) -> &[Upvar]`). 

But I won't hold up the PR on it. 

r=me, depending on how you feel @eddyb